aim/storage/rockscontainer.py

```diff
@@ -0,0 +1,489 @@
+        the `prefix`.
+
+        For example, if `prefix == b'meta.'`, and the Container consists of:
+        `{
+            b'e.y': b'012',
+            b'meta.x': b'123',
+            b'meta.z': b'x',
+            b'zzz': b'oOo'
+        }`, the method will yield `b'meta.x'` and `b'meta.z'`
+
+        Args:
+            prefix (:obj:`bytes`): the prefix that defines the key range
+        """
+        # TODO return KeyView, not iterator
+        it: Iterator[Tuple[bytes, bytes]] = self.db.iterkeys()
+        it.seek(prefix)
+        for key in it:
+            if not key.startswith(prefix):
+                break
+            yield key
+
+    def values(
+        self,
+        prefix: bytes = b''
+    ):
+        """Iterate over all the values in the given prefix key range.
+
+        The iteration is always performed in lexiographic order w.r.t keys.
+        If `prefix` is provided, iterate only over those record values that have
+        key starting with the `prefix`.
+
+        For example, if `prefix == b'meta.'`, and the Container consists of:
+        `{
+            b'e.y': b'012',
+            b'meta.x': b'123',
+            b'meta.z': b'x',
+            b'zzz': b'oOo'
+        }`, the method will yield `b'123'` and `b'x'`
+
+        Args:
+            prefix (:obj:`bytes`): the prefix that defines the key range
+        """
+        raise NotImplementedError
+
+    def update(
+        self
+    ) -> None:
+        raise NotImplementedError
+
+    def view(
+        self,
+        prefix: bytes = b''
+    ) -> 'Container':
+        """Return a view (even mutable ones) that enable access to the container
+        but with modifications.
+
+        Args:
+            prefix (:obj:`bytes`): the prefix that defines the key range of the
+                view-container. The resulting container will share an access to
+                only records in the `prefix` key range, but with `prefix`-es
+                stripped from them.
+
+                For example, if the Container contents are:
+                `{
+                    b'e.y': b'012',
+                    b'meta.x': b'123',
+                    b'meta.z': b'x',
+                    b'zzz': b'oOo'
+                }`, then `container.view(prefix=b'meta.')` will behave (almost)
+                exactly as an Container:
+                `{
+                    b'x': b'123',
+                    b'z': b'x',
+                }`
+        """
+        return PrefixView(prefix=prefix, container=self)
+
+    def batch(
+        self
+    ) -> aimrocks.WriteBatch:
+        """Creates a new batch object to store operations in before executing
+        using :obj:`RocksContainer.commit`.
+
+        The operations :obj:`RocksContainer.set`, :obj:`RocksContainer.delete`,
+        :obj:`RocksContainer.delete_range` are supported.
+
+        See more at :obj:`RocksContainer.commit`
+        """
+        return aimrocks.WriteBatch()
```

Comment:
here we expose rocksdb-specific object to outer world. maybe it will make sense to return some wrapper instead 🤔 ?

Reply:
I do understand the batch is just passed to set/delete methods and commit, but it's still accessible to "outer world"

Reply:
Right now we implicitly define the `protocol` in `Container` docs:
https://github.com/aimhubio/aim/blob/cdc0360568af879d52c1e332f1d1ca538ef00f42/aim/storage/container.py#L112-L118
Not sure about wrapping the batch object. Instead, we can define the interface / protocol, and annotate the return type of `Container.batch()` and the argument of `Container.commit()` with that.

Reply:
same here